fluidreview/api.py

```diff
@@ -235,3 +236,29 @@ def list_users():
             yield fluid_user
         next_page = urlparse(response['next']).query
         url = 'users?{}'.format(next_page) if next_page else None
+
+
+def post_payment(order):
+    """
+    Update amount paid by a user for a class when an order is fulfilled.
+    """
+    if order.status != Order.FULFILLED:
+        return
+    klass = order.get_klass()
+    user = order.user
+    if not klass or klass.bootcamp.legacy:
+        return
+    total_paid = Line.total_paid_for_klass(order.user, klass.klass_key).get('total') or Decimal('0.00')
+    payment_metadata = {
+        'value': '{:0.2f}'.format(total_paid)
+    }
+    webhook = WebhookRequest.objects.filter(user_id=user.profile.fluidreview_id, award_id=klass.klass_key).last()
+    response = FluidReviewAPI().put(
+        'submissions/{}/metadata/{}/'.format(webhook.submission_id, settings.FLUIDREVIEW_AMOUNTPAID_ID),
+        data=payment_metadata
+    )
+    if response.status_code != 200:
```

Comment:
Instead of checking the status code can you rewrite like this:

    try:
      response = FluidReviewAPI().put(...)
      response.raise_for_status()
    except Exception as ex:
        raise FluidReviewException(...) from ex

This way you won't need to include `response.json()` in the exception, it will exist in the `HTTPError` exception in the exception chain

Reply:
✅